File: Entities/Historical.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace aliment_backend.Entities
{
    /// <summary>
    /// Représente un enregistrement d'historique pour une entité Stock.
    /// </summary>
    public class Historical
    {
        /// <summary>
        /// Clé primaire de l'enregistrement historique.
        /// </summary>
        [Key]
        public int Id { get; set; }

        /// <summary>
        /// Date du contrôle historique.
        /// </summary>
        public DateTime ControleDate { get; set; }

        /// <summary>
        /// Action effectuée lors du contrôle historique (optionnel).
        /// </summary>
        public string? Action { get; set; }

        /// <summary>
        /// Clé étrangère vers l'entité Stock associée à cet historique.
        /// </summary>
        [ForeignKey("Stock")]
        public int StockId { get; set; }

        /// <summary>
        /// Référence à l'entité Stock associée à cet historique.
        /// </summary>
        public Stock? Stock { get; set; }

        /// <summary>
        /// Constructeur de la classe Historical.
        /// </summary>
        /// <param name="controleDate">Date du contrôle historique.</param>
        /// <param name="action">Action effectuée lors du contrôle historique (optionnel).</param>
        /// <param name="stockId">Identifiant de l'entité Stock associée à cet historique.</param>
        public Historical(DateTime controleDate, string? action, int stockId)
        {
            ControleDate = controleDate;
            Action = action;
            StockId = stockId;
        }
    }
}

File: src/app/features/admin/child/_model/historical.ts
import { Stock } from "../../../../shared/module/stock/_model/stock";

/**
 * La classe `Historical` représente un enregistrement historique d'une action effectuée dans l'application.
 * Elle stocke des informations sur une opération passée, comme la date, l'action effectuée, et les détails du stock associé.
 */
export class Historical {
  /**
   * Identifiant unique de l'enregistrement historique.
   * @type {number}
   */
  id!: number;

  /**
   * Date à laquelle l'action historique a été effectuée.
   * @type {Date}
   */
  controleDate!: Date;

  /**
   * Action effectuée dans l'enregistrement historique, par exemple, "Ajout", "Modification", "Suppression", etc.
   * @type {string}
   */
  action!: string;

  /**
   * Objet représentant le stock associé à cet enregistrement historique.
   * @type {Stock}
   */
  stock!: Stock;

  /**
   * Identifiant unique du stock associé à cet enregistrement historique.
   * @type {number}
   */
  stockId!: number;
}
